ClimbingGearBackend/Controllers/RackGearController.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ClimbingGearBackend.Models;
+using ClimbingGearBackend.Interfaces;
+
+namespace ClimbingGearBackend.Controllers
+{
+  [Route("api/rack/{rackId}/[controller]")]
+  [ApiController]
+  public class RackGearController : ControllerBase
+  {
+    private readonly IRackRepository _repository;
+
+    public RackGearController(IRackRepository repository)
+    {
+      _repository = repository;
+    }
+
+    [HttpGet("{gearId}")]
+    public async Task<ActionResult<RackGearDTO>> GetRackGear(long rackId, long gearId)
+    {
+      var gear = await _repository.GetRackGearByIdsAsync(rackId, gearId);
+
+      if (gear == null)
+      {
+        return NotFound();
+      }
+
+      return RackGearToRackGearDTO(gear);
+    }
+
+    [HttpPut("{gearId}")]
+    public async Task<IActionResult> UpdateRackGear(long rackId, long gearId, RackGear putRackGear)
```

Comment:
Since `RackGear` already contains the Gear Id, is it necessary to have it in the route? This could be just `PUT /api/rack/{rackId}/gear`.